libcontainer/process_linux.go

```diff
@@ -255,6 +255,19 @@ func (p *initProcess) execSetns() error {
 	return nil
 }
 
+// parentError attempts to read any error from the parent so we have better context
+// about the cause of the failure. If no error is found then a new system error for
+// err and cause is returned.
+func parentError(pipe *os.File, err error, cause string) error {
+	if ierr := parseSync(pipe, func(sync *syncT) error { return nil }); ierr != nil {
```

Comment:
Hmm, not sure we're on the same page yet.

I could manually read the pipe, slurping data until EOF looking for an procError but parseSync achieves just this, so it seems sensible to make use of this; did I miss something?

I can confirm that when init failed, I'd not included nsenter so the config read was failing with a parse error, this change correctly receives and outputs the error returned from init.